consensus/state.go

```diff
@@ -1779,16 +1779,23 @@ func (cs *State) addProposalBlockPart(msg *BlockPartMessage, peerID p2p.ID) (add
 	if err != nil {
 		return added, err
 	}
+	if cs.ProposalBlockParts.ByteSize() > cs.state.ConsensusParams.Block.MaxBytes {
```

Comment:
Could the same thing be achieved by doing?
```
reader := io.LimitReader(
cs.ProposalBlockParts.GetReader(),
cs.state.ConsensusParams.Block.MaxBytes)
bz, err := ioutil.ReadAll(reader)
```

Reply:
Is this more performant?

Reply:
I am not sure if its more performant but causes fewer changes and I believe mimics the previous behavior. I would have to check what amino was doing.